### ISSUE TITLE
[Bug]: Location doesn't change when use setLocation() method Android

### ISSUE THREAD

**Issue by Phoenix124** [appium/appium#17673]
## The problem

Try to set Location with `driver.setLocation(new Location(-81.5812,28.4194, 10));` but location doesn't change
try also to change location using adb
```
root@9dc04fa486f2:/# adb emu geo fix "-81.5812 28.4194"
OK
root@9dc04fa486f2:/# adb shell am broadcast -n io.appium.settings/.receivers.LocationInfoReceiver -a io.appium.settings.location
Broadcasting: Intent { act=io.appium.settings.location flg=0x400000 cmp=io.appium.settings/.receivers.LocationInfoReceiver }
```

## Environment

* Appium version (or git revision) that exhibits the issue: 1.23.2
* Last Appium version that did not exhibit the issue (if applicable): 1.23.2
* Desktop OS/version used to run Appium: Linux
* Node.js version (unless using Appium.app|exe): 16
* Npm or Yarn package manager: 8
* Mobile platform/version under test: Android 10
* Real device or emulator/simulator: emulator
* Appium CLI or Appium.app|exe: Appium Cli

**Comment by mykola-mokhnach** [comment]
If the location cannot be changed with adb then Appium won't be able to change it as well. 
Also Appium 1 is not maintained anymore. Please use the most recent driver version available as part of Appium 2

**Comment by Phoenix124** [comment]
@mykola-mokhnach could be a problem of avd image?
I use avd with google_apis without play_services
How could i debug a problem?

**Comment by Phoenix124** [comment]
try with appium 2.0
```
[HTTP] --> POST /wd/hub/session/18c99787-123c-45f7-957d-f1a638c8a8e2/location
[HTTP] {"location":{"longitude":37.6173,"latitude":55.755826}}
[debug] [AndroidUiautomator2Driver@da8c (18c99787)] Calling AppiumDriver.setGeoLocation() with args: [{"longitude":37.6173,"latitude":55.755826},"18c99787-123c-45f7-957d-f1a638c8a8e2"]
[debug] [ADB] Running '/var/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 emu geo fix 37.6173 55.75583'
[debug] [ADB] Running '/var/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 emu geo fix 37,6173 55,75583'
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running '/var/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Running '/var/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 shell am broadcast -n io.appium.settings/.receivers.LocationInfoReceiver -a io.appium.settings.location'
[debug] [ADB] Got geo coordinates: {"latitude":"37.4219983","longitude":"-122.0840000","altitude":"0.0000000"}
[debug] [AndroidUiautomator2Driver@da8c (18c99787)] Responding to client with driver.setGeoLocation() result: {"latitude":37.4219983,"longitude":-122.084,"altitude":5e-324}
[HTTP] <-- POST /wd/hub/session/18c99787-123c-45f7-957d-f1a638c8a8e2/location 200 17534 ms - 72
[HTTP] 
[HTTP] --> GET /wd/hub/session/18c99787-123c-45f7-957d-f1a638c8a8e2/location
[HTTP] {}
[debug] [AndroidUiautomator2Driver@da8c (18c99787)] Calling AppiumDriver.getGeoLocation() with args: ["18c99787-123c-45f7-957d-f1a638c8a8e2"]
[debug] [ADB] Getting IDs of all 'io.appium.settings' processes
[debug] [ADB] Running '/var/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 shell pgrep -f \(\[\[:blank:\]\]\|\^\)io\.appium\.settings\(\[\[:blank:\]\]\|\$\)'
[debug] [ADB] Running '/var/android-sdk/platform-tools/adb -P 5037 -s emulator-5554 shell am broadcast -n io.appium.settings/.receivers.LocationInfoReceiver -a io.appium.settings.location'
[debug] [ADB] Got geo coordinates: {"latitude":"37.4219983","longitude":"-122.0840000","altitude":"0.0000000"}
[debug] [AndroidUiautomator2Driver@da8c (18c99787)] Responding to client with driver.getGeoLocation() result: {"latitude":37.4219983,"longitude":-122.084,"altitude":5e-324}
[HTTP] <-- GET /wd/hub/session/18c99787-123c-45f7-957d-f1a638c8a8e2/location 200 6822 ms - 72
[HTTP]
```
but location not changed

**Comment by mykola-mokhnach** [comment]
UIA2 driver is just proxying the calls to adb. Consider reporting the issue to Google instead.
